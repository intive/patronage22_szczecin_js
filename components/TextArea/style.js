import styled from 'styled-components'
import TextareaAutosize from 'react-textarea-autosize'

export const StyledTextareaAutosize = styled(TextareaAutosize)`
  width: 100%;
  min-height: 80px;
  padding: 0.75rem 1rem;
  color: #3B3B3B;
  background: #F7F7F7;
  border: 1px solid #E1E1E1;
  border-radius: 8px;
  box-sizing: border-box;
  resize: none;
  outline: none;
  overflow: hidden;
`
