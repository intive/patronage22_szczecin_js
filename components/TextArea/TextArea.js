import { forwardRef } from 'react'
import { StyledTextareaAutosize } from './style'
import CharacterCounter from '../CharacterCounter/CharacterCounter'

// eslint-disable-next-line react/display-name
const TextArea = forwardRef((props, ref) => {
  return (
    <>
      <StyledTextareaAutosize ref={ref} onChange={props.onChange} maxLength={props.maxLength} />
      {props.maxLength ? <CharacterCounter inputValue={props.value} maxCharacters={props.maxLength} /> : null}
    </>
  )
})

export default TextArea
