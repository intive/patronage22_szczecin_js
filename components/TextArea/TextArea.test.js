/* globals describe, expect, it */
/* eslint-env jest */

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import TextArea from './TextArea'

describe('TextArea', () => {
  it('should render TextArea component', () => {
    const { container } = render(<TextArea />)

    expect(container).toMatchSnapshot()
  })

  it('should have value hello world and be called 13 times after onchange is called', () => {
    const mockOnChange = jest.fn()
    render(<TextArea onChange={mockOnChange} />)
    userEvent.type(screen.getByRole('textbox'), 'Hello,{enter}World!')

    expect(screen.getByRole('textbox')).toHaveValue('Hello,\nWorld!')
    expect(mockOnChange).toHaveBeenCalledTimes(13)
  })
})
