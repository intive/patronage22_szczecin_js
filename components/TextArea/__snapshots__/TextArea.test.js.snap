// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render TextArea component 1`] = `
<div>
  <textarea
    class="sc-bdvvtL dVstcH"
  />
</div>
`;
